sql/core/src/main/java/org/apache/spark/sql/vectorized/ColumnarArray.java

```diff
@@ -46,7 +47,33 @@ public int numElements() {
 
   @Override
   public ArrayData copy() {
```

Comment:
Here, I think that ColumnarArray.cope() should return a ColumnarArray. Or it should be renamed copyForGenericArrayData.

Reply:
It’s overrided  methods, I’m not sure that I can change a signature of this method.

Reply:
Why do we need to support this copy() here? Is this useful for something?

Reply:
I just found this [improvement](https://issues.apache.org/jira/browse/SPARK-25713?jql=project%20%3D%20SPARK%20AND%20issuetype%20%3D%20Improvement%20AND%20status%20%3D%20Open%20AND%20resolution%20%3D%20Unresolved%20AND%20assignee%20in%20(EMPTY)%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC)

Reply:
So, I just want to know why we need the improvement.

Reply:
previously this method throw 'UnsupportedOperationException' and I guess that it's not correct behaving of override methods.

Reply:
The only problem I see with this is that it doesn't return a ColumnarArray. I'm not sure if that's possible as ColumnVector doesn't implement a clone() or copy() method, and maybe I'm missing reasons that's not sensible. So this might be intentional. I don't see any rationale from whoever opened the JIRA

Reply:
@srowen, does it mean that I need to close pull request?

Reply:
You could ping the people who wrote this code, maybe @cloud-fan .. if there's no affirmative reply in some time, yeah I'd leave this as-is.

Reply:
I think this is a public API ... should be okay to fix